micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/custom/ResponseRethrowingErrorHandler.groovy

```diff
@@ -16,6 +16,29 @@ import org.springframework.web.client.ResponseErrorHandler
 @CompileStatic
 class ResponseRethrowingErrorHandler implements ResponseErrorHandler {
 
+    static final enum ErrorMessageLogging {
+        NONE, WARN, ERROR
+    }
+
+    private final ErrorMessageLogging errorMessageLogging
```

Comment:
I prefer `Optional`, because if I set the default value to something, I'd have to also change the `null` in RestTemplate calling this constructor `new ReponseRethrowingErrorHandler(null)` - with `Optional` I'll have better control over what default behavior is.